### PR TITLE
fix: match filter, prefer filter_fn and fix style

### PR DESCRIPTION
**Pull request by nomagick**
Fix match filter (bool(x) => bool(x.text))
Prefer filter_fn over loop-continue
Clean commented out code
Apply black with single quote style fix
Migrate git hooks, formatting, linting from executors
Add formatter options for vscode

### CODE REVIEW

File: dpr_reader.py [hunk 1]
```diff
@@ -105,26 +107,33 @@ def rank(
...
-            # match_batches_generator = doc_arr.batch(
-            #     traversal_paths=['m'],
-            #     batch_size=batch_size,
-            #     require_attr='text',
-            # )
-            match_batches_generator = doc_arr.traverse_flat(traversal_paths=['m'], filter_fn= lambda x: bool(x)).batch(batch_size=batch_size)
+
+            match_batches_generator = doc_arr.traverse_flat(
+                traversal_paths=['m'],
+                filter_fn=lambda x: bool(x.text)
```
Comment (nomagick) [inline dpr_reader.py]: Is this filter function identical to `require_attr='text'` ?

Comment (tadejsv) [reply dpr_reader.py]: Looks equivalent yes

File: dpr_reader.py [hunk 2]
```diff
@@ -105,26 +107,33 @@ def rank(
...
+            ).batch(batch_size=batch_size)
+
             for matches in match_batches_generator:
-                question, titles = self._prepare_inputs(doc_arr[0].text, matches)
+                question, titles = self._prepare_inputs(
+                    doc_arr[0].text,
```
Comment (tadejsv) [inline dpr_reader.py]: Why do we break lines here now?

File: dpr_reader.py [hunk 3]
```diff
@@ -105,26 +107,33 @@ def rank(
...
+                    matches
+                )
                 with torch.inference_mode():
-                    new_matches += self._get_new_matches(question, matches, titles)
+                    new_matches += self._get_new_matches(
+                        question,
```
Comment (tadejsv) [inline dpr_reader.py]: Same here

File: dpr_reader.py (hunk 2 above)
Comment (tadejsv) [reply dpr_reader.py]: We use `black` for styling, not autpep8

Comment (nomagick) [reply dpr_reader.py]: I see. I should also migrate git hooks from the executors.

**Pull request merged**